### PR TITLE
Add .nuspec for Elmah.AspNet to publish as a package

### PR DESCRIPTION
**Pull request by paal123**
Necessary code to pack the Elmah.AspNet nuget package for alpha 001.

### CODE REVIEW

File: pkg/Elmah.AspNet.nuspec [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        <title>ELMAH AspNet Library</title>
+        <authors>Atif Aziz</authors>
+        <owners>Atif Aziz, James Driscoll</owners>
+        <licenseUrl>http://www.apache.org/licenses/LICENSE-2.0</licenseUrl>
+        <projectUrl>https://elmah.github.io/</projectUrl>
+        <requireLicenseAcceptance>false</requireLicenseAcceptance>
+        <description>ELMAH (Error Logging Modules and Handlers) AspNet.</description>
+        <summary>ELMAH (Error Logging Modules and Handlers) AspNet without Elmah.CoreLibrary.</summary>
+        <language>en-US</language>
+        <tags>elmah error logging unhandled exception</tags>
```
Comment (atifaziz) [inline pkg/Elmah.AspNet.nuspec]: Add [`aspnet`][aspnet] and [`asp.net`][asp.net] as tags too.

[aspnet]: https://www.nuget.org/packages?q=Tags%3A%22aspnet%22
[asp.net]: https://www.nuget.org/packages?q=Tags%3A%22asp.net%22

File: pkg/Elmah.AspNet.nuspec [hunk 2]
```diff
@@ -0,0 +1,39 @@
+<?xml version="1.0"?>
+<package xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
+    <metadata xmlns="http://schemas.microsoft.com/packaging/2010/07/nuspec.xsd">
+        <id>elmah.aspnet</id>
+        <version>2.0.0-alpha001</version>
+        <title>ELMAH AspNet Library</title>
```
Comment (atifaziz) [inline pkg/Elmah.AspNet.nuspec]: This should read simply “ELMAH for ASP.NET”

File: pkg/Elmah.AspNet.nuspec [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        <authors>Atif Aziz</authors>
+        <owners>Atif Aziz, James Driscoll</owners>
+        <licenseUrl>http://www.apache.org/licenses/LICENSE-2.0</licenseUrl>
+        <projectUrl>https://elmah.github.io/</projectUrl>
+        <requireLicenseAcceptance>false</requireLicenseAcceptance>
+        <description>ELMAH (Error Logging Modules and Handlers) AspNet.</description>
```
Comment (atifaziz) [inline pkg/Elmah.AspNet.nuspec]: “ELMAH (Error Logging Modules and Handlers) for ASP.NET” reads better.

File: pkg/Elmah.AspNet.nuspec [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+        <summary>ELMAH (Error Logging Modules and Handlers) AspNet without Elmah.CoreLibrary.</summary>
```
Comment (atifaziz) [inline pkg/Elmah.AspNet.nuspec]: This can be removed. IIRC, the [`<summary>` tag is being deprecated](https://github.com/NuGet/Home/issues/4587#issuecomment-288913324) and the description is short and sweet enough.

File: Elmah.sln [hunk 5]
```diff
@@ -1,6 +1,6 @@
 Microsoft Visual Studio Solution File, Format Version 12.00
-# Visual Studio 2013
-VisualStudioVersion = 12.0.31101.0
+# Visual Studio 15
```
Comment (atifaziz) [inline Elmah.sln]: This is an unrelated change. Can we get rid of it?

File: Elmah.sln [hunk 6]
```diff
@@ -107,4 +108,7 @@ Global
 	GlobalSection(SolutionProperties) = preSolution
 		HideSolutionNode = FALSE
 	EndGlobalSection
+	GlobalSection(ExtensibilityGlobals) = postSolution
```
Comment (atifaziz) [inline Elmah.sln]: This is an unrelated change. Can we get rid of it?

Comment (paal123) [reply Elmah.sln]: Removed

File: pkg/Elmah.AspNet.nuspec (hunk 2 above)
Comment (paal123) [reply pkg/Elmah.AspNet.nuspec]: Changed

File: pkg/Elmah.AspNet.nuspec (hunk 1 above)
Comment (paal123) [reply pkg/Elmah.AspNet.nuspec]: added

File: pkg/Elmah.AspNet.nuspec [hunk 7]
```diff
@@ -2,17 +2,16 @@
 <package xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
     <metadata xmlns="http://schemas.microsoft.com/packaging/2010/07/nuspec.xsd">
         <id>elmah.aspnet</id>
-        <version>2.0.0-alpha001</version>
-        <title>ELMAH AspNet Library</title>
+        <version>2.0.0-alpha002</version>
```
Comment (atifaziz) [inline pkg/Elmah.AspNet.nuspec]: What did you have in mind with this pre-release version bump?

Comment (paal123) [reply pkg/Elmah.AspNet.nuspec]: Adjusted

File: pkg/Elmah.AspNet.nuspec [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+        <tags>elmah aspnet asp.net error logging unhandled exception</tags>
+        <dependencies>
+          <dependency id="elmah.corelibrary" version="2.0.0-alpha001" />
+        </dependencies>
+    </metadata>
+    <files>
+        <file src="..\bin\net-3.5\Release\Elmah.AspNet.dll" target="lib\net35" />
+        <file src="..\bin\net-3.5\Release\Elmah.AspNet.pdb" target="lib\net35" />
+        <file src="..\bin\net-3.5\Release\Elmah.AspNet.xml" target="lib\net35" />
+        <file src="..\bin\net-3.5\Release\AntiXssLibrary.dll" target="lib\net35" />
```
Comment (atifaziz) [inline pkg/Elmah.AspNet.nuspec]: The `AntiXssLibrary.dll` doesn't belong in this package. Instead it should be pulled in as a dependency on [AntiXSS 4.2.1]( https://www.nuget.org/packages/AntiXSS/4.2.1).

**Comment by atifaziz** [comment]
@paal123 I couldn't push the package for the life of me and then I realized after getting some permission errors that the [elmah.aspnet](https://www.nuget.org/packages/elmah.aspnet/) package was already published by you, _a day ago_! Since that was before this merge, it's not reflective of what went into the master with 516a2b074cf7e382160a21322bed41b1565d2931. For example, it's missing the dependency on AntiXss that was added in fd00282ff12d2c0b2b1afa603da487139c38ab10.

I don't think a package can be deleted, so unfortunately, we'll have to publish a new alpha. Is it possible for you to yield the ownership of the **elmah.aspnet** package?

**Comment by paal123** [comment]
I have added you now @atifaziz !

**Comment by atifaziz** [comment]
@paal123 Thanks for adding me as an owner. Going forward, I think it's important to release packages on NuGet through a more formal/coordinated process so that we can avoid the current situation where [2.0.0-alpha001](https://www.nuget.org/packages/elmah.aspnet/2.0.0-alpha001) and [2.0.0-alpha002](https://www.nuget.org/packages/elmah.aspnet/2.0.0-alpha002) are not in sync with the code base, even if they are alpha. If you need to work with bleeding builds then we need to setup CI and use the [elmah package feed at myget.org](https://www.myget.org/feed/Packages/elmah).
